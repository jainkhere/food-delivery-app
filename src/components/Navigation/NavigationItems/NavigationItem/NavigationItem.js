import React from "react";
import "./NavigationItem.css";
import PropTypes from "prop-types";

const navigationItem = (props) => {
  const { children, link, active, } = props;
  return (
    <li className="navigationItem">
      <a href={link} className={active ? "active" : null}>
        {children}
      </a>
    </li>
  );
};

navigationItem.propTypes = {
  children: PropTypes.node.isRequired,
  link: PropTypes.string.isRequired,
  active: PropTypes.bool.isRequired,
};

export default navigationItem;
