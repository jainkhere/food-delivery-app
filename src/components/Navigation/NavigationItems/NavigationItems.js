import React from "react";
import NavigationItem from "./NavigationItem/NavigationItem";
import "./NavigationItems.css";

const navigationItems = () => (
  <ul className="navigationItems">
    <NavigationItem link="/mobile-first-react-hamburger" active>
      Burger Builder
    </NavigationItem>
  </ul>
);

export default navigationItems;
