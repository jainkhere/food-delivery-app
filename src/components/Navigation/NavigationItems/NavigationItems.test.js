import Adapter from "enzyme-adapter-react-16";
import React from "react";
import Enzyme, { shallow, } from "enzyme";

import NavigationItems from "./NavigationItems";
import NavigationItem from "./NavigationItem/NavigationItem";

Enzyme.configure({ adapter: new Adapter(), });

describe("<NavigationItems />", () => {
  it("should render 1 <NavigationItem /> element", () => {
    const wrapper = shallow(<NavigationItems />);
    expect(wrapper.find(NavigationItem)).toHaveLength(1);
  });
});
