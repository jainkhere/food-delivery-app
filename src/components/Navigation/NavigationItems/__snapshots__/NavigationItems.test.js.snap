// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItems /> should render 1 <NavigationItem /> element 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <navigationItems />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <navigationItem
          active={true}
          link="/mobile-first-react-hamburger"
>
          Burger Builder
</navigationItem>,
        "className": "navigationItems",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active": true,
          "children": "Burger Builder",
          "link": "/mobile-first-react-hamburger",
        },
        "ref": null,
        "rendered": "Burger Builder",
        "type": [Function],
      },
      "type": "ul",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <navigationItem
            active={true}
            link="/mobile-first-react-hamburger"
>
            Burger Builder
</navigationItem>,
          "className": "navigationItems",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": true,
            "children": "Burger Builder",
            "link": "/mobile-first-react-hamburger",
          },
          "ref": null,
          "rendered": "Burger Builder",
          "type": [Function],
        },
        "type": "ul",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
      "attachTo": undefined,
      "hydrateIn": undefined,
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "active": true,
      "children": "Burger Builder",
      "link": "/mobile-first-react-hamburger",
    },
    "ref": null,
    "rendered": "Burger Builder",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": true,
        "children": "Burger Builder",
        "link": "/mobile-first-react-hamburger",
      },
      "ref": null,
      "rendered": "Burger Builder",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
